pkg/readiness/readiness.go

```diff
@@ -60,30 +59,48 @@ func Run(bird, bird6, felix bool) {
 // checkBIRDReady checks if BIRD is ready by connecting to the BIRD
 // socket to gather all BGP peer connection status, and overall graceful
 // restart status.
-func checkBIRDReady(ipv string) error {
+func checkBIRDReady(ipv string, thresholdTime time.Duration) error {
 	// Check for unestablished peers
 	peers, err := bird.GetPeers(ipv)
 	log.Debugf("peers: %v", peers)
 	if err != nil {
 		return err
 	}
 
+	// numEstablishedPeer keeps count of number of peers with bgp state established
```

Comment:
How about moving the Stat() up to this point itn he function, then you can move the rest of the function to a new utility function, which can be easily UTed.

I.e. the new utility function should take arguments for the BGP peers and the node startup time.